micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmHeapPressureMetrics.java

```diff
@@ -78,10 +78,14 @@ public void bindTo(@NonNull MeterRegistry registry) {
                 .tag("area", "heap")
                 .description("The percentage of old gen heap used after the last GC event, in the range [0..1]")
                 .baseUnit(BaseUnits.PERCENT);
-        if (JvmMemory.isOldGenPool(longLivedPoolName))
-            builder.tag("generation", "old");
-        else
-            builder.tag("pool", longLivedPoolName);
+
+        if ( longLivedPoolName != null ) {
```

Comment:
If we can't get the long-lived pool name, then we can't calculate the `jvm.memory.usage.after.gc` and might as well not register it. As is, it will always be zero since it is never updated (the update is guarded by a null-check on `longLivedPoolName`).